### PR TITLE
added gitignore for firebase

### PR DESCRIPTION
**Pull request by DrDead0**
**Reasons for making this change:**  
As a developer working with Firebase, it is essential to exclude sensitive files, auto-generated content, and irrelevant build artifacts from version control to enhance security and streamline repository management. This `.gitignore` template is tailored to Firebase projects, ensuring best practices for file exclusion.

This change aims to provide the Firebase development community with a pre-configured `.gitignore` file, saving time and reducing potential errors.

**Links to documentation supporting these rule changes:**  
- [Firebase Documentation - Sensitive Configuration](https://firebase.google.com/docs/projects/learn-more#sensitive-data)
- [Git Documentation - Ignoring Files](https://git-scm.com/docs/gitignore) 
- [Best Practices for Firebase Security Rules](https://firebase.google.com/docs/rules)

If this is a new template:  
- **Link to application or project’s homepage**: [Firebase Official Homepage](https://firebase.google.com/)

### CODE REVIEW

File: Firebase.gitignore [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+yarn-error.log*
+
+# OS-specific files
+.DS_Store
+Thumbs.db
+
+# IDE files
+.idea/
+.vscode/
+*.sublime*
```
Comment (thorrsson) [inline Firebase.gitignore:35]: @DrDead0 per https://github.com/github/gitignore?tab=readme-ov-file#contributing-guidelines

```
Please consider the scope of your change. If your change is specific to a certain language or framework, then make sure the change is made to the template for that language or framework, rather than to the template for an editor, tool, or operating system.
```

Could you remove these lines that are not related to firebase itself and are Editor and OS specific?

File: Firebase.gitignore [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+# Environment files (local configs)
+.env
+.env.*
+.env.local
+.env.development
+.env.production
+.env.test
+.env.staging
```
Comment (thorrsson) [inline Firebase.gitignore:44]: These lines are not necessary given the `.env.*` regex above would actually catch them as well.
I'd also argue that these are not Firebase specific files given `.env` files can be used with JS, Ruby, Python, etc so we may want to remove them entirely as well

**Review by thorrsson — COMMENTED** [review]
@DrDead0 thanks very much for this contribution!
I added a couple of comments for you

**Comment by DrDead0** [comment]
Thanks for the feedback! I've updated the `.gitignore` file to:
- Remove IDE and OS-specific lines, as they are unrelated to Firebase.
- Simplify `.env` patterns by retaining only `/env.*`.
- Scope the template strictly to Firebase-related files.

Let me know if further adjustments are needed!

**Comment by Drew810** [comment]
> 
> 
> 
> If this is a new template:  
> 
> - **Link to application or project’s homepage**: [Firebase Official Homepage](https://firebase.google.com/)
> 
> 
> 
>

File: Firebase.gitignore [hunk 3]
```diff
@@ -0,0 +1,28 @@
+# Firebase build and deployment files
+/firebase-debug.log
```
Comment (Drew810) [inline Firebase.gitignore:2]: "[ms-outlook://people/v3/0009002e00000000001d840311aa6611cd9bc800aa002fc45a0d002e485adcecc36d47b2186bc9c3649fc2000558638ba10000?a=67c9404f37f131d608e8726c33899fdd4068f290fea72b82fd855970c949524c&c=ba1a0757

```suggestion
/firebase-debug.log



```
]()"}

**Review by Drew810 — CHANGES_REQUESTED** [review]

**Review by thorrsson — APPROVED** [review]
Thanks for the contribution @DrDead0
